org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/launch/internal/GradleClasspathProvider.java

```diff
@@ -56,7 +57,32 @@ public GradleClasspathProvider() {
 
     @Override
     public IRuntimeClasspathEntry[] computeUnresolvedClasspath(ILaunchConfiguration configuration) throws CoreException {
-        return super.computeUnresolvedClasspath(configuration);
+        return filterUnusedDependencies(configuration, super.computeUnresolvedClasspath(configuration));
+    }
+
+    private IRuntimeClasspathEntry[] filterUnusedDependencies(ILaunchConfiguration configuration, IRuntimeClasspathEntry[] entriesToFilter) throws CoreException {
+        // if the run configuration uses Java 9 then the library dependencies are already present in the
```

Comment:
What does this mean in practice, i.e. who uses the unresolved classpath?

Reply:
JDT itself. If the target platform is Java 9 then the unresolved classpath already contains the jar dependencies.